app/frontend/actions.ts

```diff
@@ -603,17 +603,25 @@ export default ({setState, getState}: {setState: SetStateFn; getState: GetStateF
 
   const prepareTxPlan = async (args) => {
     const state = getState()
-    const plan: any = await wallet.getAccount(state.sourceAccountIndex).getTxPlan(args)
-    // FIXME: this cant be any
-    if (plan.error) {
-      stopLoadingAction(state, {})
-      resetDelegationWithoutHash(state)
-      setState({
-        calculatingDelegationFee: false,
-        calculatingFee: false,
-      })
+    try {
+      const plan: any = await wallet.getAccount(state.sourceAccountIndex).getTxPlan(args)
+      // FIXME: this cant be any
+      if (plan.error) {
+        stopLoadingAction(state, {})
+        resetDelegationWithoutHash(state)
+        setState({
+          calculatingDelegationFee: false,
+          calculatingFee: false,
+        })
+      }
+      return plan
+    } catch (e) {
+      if (e.name !== 'NetworkError' || e.name !== 'ServerError') throw e
```

Comment:
I would rather wrap the statement in curly braces so the return statement right below doesn't get confused as the logic triggered by the condition

```suggestion
      if (e.name !== 'NetworkError' || e.name !== 'ServerError') {
        throw e
      }

```